miniconf_derive/src/lib.rs

```diff
@@ -252,11 +252,22 @@ pub fn derive_tree_deserialize(input: TokenStream) -> TokenStream {
     let depth = fields.iter().fold(0usize, |d, field| d.max(field.depth)) + 1;
     let ident = input.ident;
 
-    let (impl_generics, ty_generics, where_clause) = input.generics.split_for_impl();
+    let orig_generics = input.generics.clone();
+    let (_, ty_generics, where_clause) = orig_generics.split_for_impl();
+    let lts: Vec<_> = input.generics.lifetimes().cloned().collect();
+    input.generics.params.push(parse_quote!('de));
+    if let Some(syn::GenericParam::Lifetime(de)) = input.generics.params.last_mut() {
+        assert_eq!(de.lifetime.ident, "de");
+        for l in lts {
+            assert!(l.lifetime.ident != "de");
```

Comment:
We could technically loop through randomized lifetime names until we find one that is unused, but tbh I like just using `'de`. I think it's fairly unlikely that someone will use this in their type, and if so, we catch it here.